AddInstallationScriptTest.cmake

```diff
@@ -0,0 +1,25 @@
+macro(add_installation_script_test name path)
+
+  # Copy the source to the binary tree 
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/${path}/${name}
+    ${CMAKE_CURRENT_BINARY_DIR}/${path}/${name}
+    COPYONLY
+  )
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/${path}/${name}-usage
+    ${CMAKE_CURRENT_BINARY_DIR}/${path}/${name}-usage
+    COPYONLY
+  )
+  set(harness "${CMAKE_BINARY_DIR}/staging/test-${name}.sh")
+  install(
+      FILES "${harness}"
+      PERMISSIONS WORLD_EXECUTE WORLD_READ WORLD_WRITE OWNER_EXECUTE OWNER_READ OWNER_WRITE GROUP_EXECUTE GROUP_READ GROUP_WRITE
+      DESTINATION ${CMAKE_BINARY_DIR}/${path}
+  )
```

Comment:
Do we want to *install* the shell script tests? Usually this sort of thing is just run at build time, along with unit tests and not installed....